src/map.h

```diff
@@ -256,13 +256,14 @@ class map
 
         void set_seen_cache_dirty( const tripoint change_location ) {
             if( inbounds_z( change_location.z ) ) {
-                level_cache cache = get_cache( change_location.z );
-                if( cache.seen_cache_dirty ) {
+                std::unique_ptr<level_cache> cache =
+                    std::make_unique<level_cache>( get_cache( change_location.z ) );
+                if( cache->seen_cache_dirty ) {
                     return;
                 }
                 if( change_location == tripoint_zero ||
-                    cache.seen_cache[change_location.x][change_location.y] != 0.0 ) {
-                    cache.seen_cache_dirty = true;
+                    cache->seen_cache[change_location.x][change_location.y] != 0.0 ) {
+                    cache->seen_cache_dirty = true;
```

Comment:
It doesn't make any sense to set this flag on the local copy of this cache which is about to be thrown away.  I'm guessing this is a bug and `cache` was supposed to have been a `level_cache&` (in which case we don't need to allocate anything on the heap).  @kevingranade and @Aivean have been working on this cache stuff lately; they might have input.

Reply:
You are probably on to something. I don't have much of an understanding of this cache system and am little more than an ape clanging away on a keyboard.

Reply:
@jbytheway correct, this is one of the bugs with `seen_cache` that I've fixed in #44093:
https://github.com/CleverRaven/Cataclysm-DDA/pull/44093/files#diff-3400d316c66c62c561be6d91267695ceR269

Perhaps we can change `level_cache` to `level_cache&` here, it will help with some slowdown caused by memory copy, however I think that `seen_cache` still won't work correctly until #44093 is merged.

Reply:
Ok now I get it. Yeah, my original code made no sense. Doh. 

Thanks for the feedback. Ran a test with the change and no problems.